src/app/util/attribute-storage.cpp

```diff
@@ -336,6 +344,16 @@ CHIP_ERROR emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, const EmberA
     emAfEndpoints[index].deviceTypeList = deviceTypeList;
     emAfEndpoints[index].endpointType   = ep;
     emAfEndpoints[index].dataVersions   = dataVersionStorage.data();
+#if CHIP_CONFIG_USE_ENDPOINT_UNIQUE_ID
+    MutableCharSpan targetSpan(emAfEndpoints[index].endpointUniqueId);
+    CopyCharSpanToMutableCharSpanWithTruncation(endpointUniqueId, targetSpan);
```

Comment:
why with truncation? Why not return error on failure and use the non-truncation version?